player/js/utils/PropertyFactory.js

```diff
@@ -79,38 +81,63 @@ function interpolateValue(frameNum, caching) {
         keyframeMetadata.__fnct = fnc;
       }
       perc = fnc((frameNum - keyTime) / (nextKeyTime - keyTime));
-      var distanceInLine = bezierData.segmentLength * perc;
 
-      var segmentPerc;
-      var addedLength = (caching.lastFrame < frameNum && caching._lastKeyframeIndex === i) ? caching._lastAddedLength : 0;
-      j = (caching.lastFrame < frameNum && caching._lastKeyframeIndex === i) ? caching._lastPoint : 0;
-      flag = true;
-      jLen = bezierData.points.length;
-      while (flag) {
-        addedLength += bezierData.points[j].partialLength;
-        if (distanceInLine === 0 || perc === 0 || j === bezierData.points.length - 1) {
-          kLen = bezierData.points[j].point.length;
-          for (k = 0; k < kLen; k += 1) {
-            newValue[k] = bezierData.points[j].point[k];
+      const startPoint = keyData.s;
+      const endPoint = nextKeyData.s || keyData.e;
+
+      if (perc > 1) {
+        let tangent = keyData.ti;
+        if (Math.hypot(...tangent) <= Number.EPSILON) {
+          tangent = subtractVectors(addVectors(startPoint, keyData.to), endPoint);
```

Comment:
Is it possible that this ends up being ~0 as well? For example, if there's no first tangent set, as in here:
![image](https://github.com/lottielab/lottie-web/assets/968875/8b651254-00ed-48d8-8f3d-b18576f85105)

Reply:
It would only be zero if both tangents are zero and the two endpoints coincide. So if there is no path at all.

Reply:
Oooor, if the path is a straight line and `keyData.to` is set to be exactly `endPoint - startPoint`. I can try to handle these edge cases too.

Reply:
Okay, I pulled in the full tangent direction computation that I wrote for Lottielab. This is as close to 100% safe as I can make it (never divides by 0, if there is no path, it will do nothing as expected). This should also fix your next comment.

Reply:
I actually misread the expression when writing the first comment... Turns out actual edge case was much more edgy than what I initially thought.

However I do think that this full solution is easier to reason about and is more "convincingly correct" to me. So I vote we keep it.